### DISCUSSION TITLE
differences between this project and git-branchless

### DISCUSSION THREAD

**Discussion by hakanalpaydd** [aviator-co/av#442] [General]
Hi does anyone have a summary with differences between `av` and `git-branchless`
https://github.com/arxanas/git-branchless

I use git-branchless but there are issues with using it with a `rush` based project because when you run `rush install` it will delete all of the git hooks installed by `git-branchless`. How does `av` work without hooks?

I see av has `av commit split` which is cool
but I dont see anything like `git move` or `git prev` or `git next` which git branchless has, so curious if those things are supported via other commands or something.

**Comment by hakanalpaydd** [comment]
![rushstack zulipchat com_](https://github.com/user-attachments/assets/24a4c298-bc6b-4f79-894e-3fba07263e6f)

**Reply by hakanalpaydd** [reply]
The above screenshot is waleed (git branchless author) talking to rush the package manager my company uses for our web code base

**Reply by hakanalpaydd** [reply]
https://rushstack.zulipchat.com/#narrow/channel/262513-general/topic/rush.20install.20deletes.20my.20git.20hooks

zulip chat signup required

**Comment by Kimeiga** [comment]
Commenting with personal account to get notifications

**Comment by jainankit** [comment]
Hi @Kimeiga @hakanalpaydd , the `av` CLI does not use the `.git/hooks` model, it just maintains some branch meta-data locally. You can use `av stack next` or `av stack prev` or even interactive `av stack switch` to switch between branches. If you want to extend native git commands, you can use aliases: https://docs.aviator.co/aviator-cli/how-to-guides/git-subcommand-aliasing


To get an overview, I'd recommend watching a very simple tutorial: https://docs.aviator.co/aviator-cli

**Reply by hakanalpaydd** [reply]
out of curiosity would there be any interest in making an ISL like sapling
https://sapling-scm.com/docs/addons/vscode/

i've found it's a lot easier to do commit splitting visually this way than via the terminal because you can see the full context

**Reply by jainankit** [reply]
Definitely interested in a VS Code extension, but likely cannot prioritize at the moment.
